src/framework/MOM_domains.F90

```diff
@@ -448,8 +448,6 @@ subroutine MOM_domains_init(MOM_dom, param_file, symmetric, static_memory, &
     call create_MOM_domain(MOM_dom_unmasked, n_global, n_halo, reentrant, tripolar_N, layout_unmasked, &
                            domain_name=domain_name, symmetric=symmetric, thin_halos=thin_halos, &
                            nonblocking=nonblocking)
-  else
-    MOM_dom_unmasked => null()
```

Comment:
this `else` is tied to `if (present(MOM_dom_unmasked) .and. mask_table_exists) then`; do you need to nullify `MOM_dom_unmasked` if it is present but `mask_table_exists` is false? If so, you could change this to

```diff
- else
+ else if (present(MOM_dom_unmasked) then
    MOM_dom_unmasked => null()
```

Instead of removing this block altogether

Reply:
No need to nullify MOM_dom_unmasked. It is already null, and remains so unless the first branch (`present(MOM_dom_unmasked) .and. mask_table_exists`) is executed.